sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -206,7 +206,7 @@ private case object MySQLDialect extends JdbcDialect with SQLConfHelper {
           val indexName = rs.getString("key_name")
           val colName = rs.getString("column_name")
           val indexType = rs.getString("index_type")
-          val indexComment = rs.getString("Index_comment")
+          val indexComment = rs.getString("index_comment")
```

Comment:
Can't get correct results with `Index_comment`? This change looks more like a bug fix?

Reply:
The column name is actually `Index_comment`, but case doesn't matter. In Java [doc](https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html), it has "Column names used as input to getter methods are case insensitive".

Reply:
eg
mysql> SHOW INDEXES FROM xxx;
+----------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table    | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+----------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| xxx |          0 | PRIMARY  |            1 | jobId       | A         |           0 |     NULL | NULL   |      | BTREE      |         |               |
+----------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+

Reply:
maybe consistency seems more natural, haha